app/models/competition.rb

```diff
@@ -1284,7 +1268,7 @@ def date_range
     ApplicationController.helpers.wca_date_range(self.start_date, self.end_date)
   end
 
-  def has_date_errors?
+  def contains_date_errors?
```

Comment:
Sneaky, `contains` is just a convoluted way of saying `has`, and that in turn is considered "banned". So you're just dancing around the cliff here.